pkg/specgen/generate/pod_create.go

```diff
@@ -33,6 +33,12 @@ func MakePod(p *entities.PodSpec, rt *libpod.Runtime) (*libpod.Pod, error) {
 		}
 		p.PodSpecGen.InfraImage = imageName
 		p.PodSpecGen.InfraContainerSpec.RawImageName = imageName
+
+		if len(p.PodSpecGen.SharedNamespaces) == 1 && p.PodSpecGen.SharedNamespaces[0] == "" {
```

Comment:
Can we do this further up the stack? Reasoning: If the user does `--share none` but also `--infra true` I think we should still create the infra container, and by this point we've lost any information on whether `--infra` was set.

Reply:
I had that at first, but then I do not think API users get this ability.

Reply:
If the API sends us a SpecGen that says an infra container should be created, we should make an infra container, even if the SpecGen as a whole does not make sense.

Reply:
ok, I'll move it up in that case